File: Ruby/BasicAlgorithms/IterateArray.rb
# Write a function that would iterate through 
# each item of the given integer array and 
# print each value to the console. 
module IterateArray

  def IterateArray.For(numArray)
    newArray = []
    for num in numArray
      newArray.push(num)
    end
    print "For Array #{newArray} \n" 
    return newArray
  end

  def IterateArray.While(numArray)
    x = 0
    newArray = []
    begin
      newArray.push(numArray[x])
      x += 1    
    end while x < numArray.length
    print "While Array #{newArray} \n" 
    return newArray
  end

  def IterateArray.DoUntil(numArray)
    x = 0
    newArray = []
    until x == numArray.length do
      newArray.push(numArray[x])
      x += 1      
    end
    print "Until Array #{newArray} \n" 
    return newArray
  end

end

File: JavaScript/BasicAlgorithms/IterateArray.js
//  Write a function that would iterate through 
//  each item of the given integer array and 
//  print each value to the console. 

export function For(numberArray)
{
    let sum = 0;
    for(var iterator = 0; iterator < numberArray.length - 1; iterator++)
    {
        sum += iterator;        
        console.log(`For Array: ${numberArray[iterator]}`);
    }
}

export function ForIn(numberArray)
{
    for(let number in numberArray)
    {     
        console.log(`For In Array: ${numberArray[number]}`);
    }    
}

export function ForOf(numberArray)
{
    for(let number of numberArray)
    {
        console.log(`For Of Array: ${number}`);     
    }    
}

export function While(numberArray)
{
    let iterator = 0;
    while(iterator < numberArray.length)
    {
        console.log(`While Array: ${numberArray[iterator]}`);
        iterator++;
    }
}

export function DoWhile(numberArray)
{
    let iterator = 0;    
    do{
        console.log(`Do While Array: ${numberArray[iterator]}`);        
        iterator++;
    }    
    while(iterator < numberArray.length)
}